### DISCUSSION TITLE
fetch method not working properly

### DISCUSSION THREAD

**Discussion by ghost** [deta/cloud-docs#519] [Help]
```py
db.fetch({'phoneNumber': '07011111111'})
```

I want to fetch with a phone number, but when I fetch with a matching value, I don't get the correct result.
0 results

**Comment by lemonyte** [comment]
Are you storing your phone numbers as strings? Try getting the item by key and compare the returned data with your fetch query.

**Comment by Sokhavuth** [comment]
You better use get() method rather than fetch() when you retrieve only one item. If you use Node.js, don't forget to use "await" keyword, otherwise, you will get a promise object.
[https://docs.deta.sh/docs/base/sdk](https://docs.deta.sh/docs/base/sdk)

```js
// get one record {key:'xxxxxx', 'phoneNumber': '07011111111'} with the key or id ''xxxxxx''
const phone_number_record = await db.get("xxxxxxx")

// get an array of one record [ {key:'xxxxxx', 'phoneNumber': '07011111111'} ]
const array_of_phone_number_record = await db.fetch({'phoneNumber': '07011111111'})
